### PR TITLE
Add a `globus flows run update` command

### PR DESCRIPTION
**Pull request by kurtmckee**
### Enhancements

* Add a `globus flows run update` command.

### CODE REVIEW

File: src/globus_cli/commands/flows/run/update.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+@command("update", short_help="Update a run")
+@run_id_arg
+@click.option(
+    "--label",
+    type=str,
+    help="A label to give the run.",
+)
+@click.option(
+    "--managers",
+    "run_managers",
```
Comment (ada-globus) [inline src/globus_cli/commands/flows/run/update.py:21]: Is there a reason why you adopted an alternate name here but not in the principals for the `flows update` command (I'm guessing we haven't aliased these at the API level...)? No objection, just double checking.

**Review by ada-globus — APPROVED** [review]

File: src/globus_cli/commands/flows/run/update.py (hunk 1 above)
Comment (kurtmckee) [reply src/globus_cli/commands/flows/run/update.py:21]: Do you mean "not in the principals for the `flow start` command"?

Flows use different terms (administrators, starters, viewers) than Runs (monitors, managers).

As for the difference with `flows start`, I thought the plan was to modify how those commands work, and asap.

Comment (ada-globus) [reply src/globus_cli/commands/flows/run/update.py:21]: Sorry, I should have been more precise: In other commands we are not using `flows_` or `run_` prefixes. I was just wondering why this one is.

File: tests/functional/flows/test_update_run.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        "Run ID",
+        "Run Label",
+        "Run Managers",
+        "Run Monitors",
+        "Run Tags",
+    }
+    for field in expected_fields:
+        assert field in result.output, f"'{field}' is not present in the output"
+
+    tag_match = re.search(r"^Run Tags:\s+(?P<tags>.+)$", result.output, flags=re.M)
```
[the remark this answers is not in the document]
Comment (ada-globus) [reply tests/functional/flows/test_update_run.py:100]: Yeah, agree, that would be a bit more intuitive!

File: src/globus_cli/commands/flows/run/update.py (hunk 1 above)
Comment (kurtmckee) [reply src/globus_cli/commands/flows/run/update.py:21]: Ah, gotcha.

I did that so it was clearer where that would map to in the API call.

Comment (kurtmckee) [reply src/globus_cli/commands/flows/run/update.py:21]: I can change that if desired.

File: tests/functional/flows/test_update_run.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sirosen) [reply tests/functional/flows/test_update_run.py:100]: Okay, I'm going to have to circle back to this. I think the `result` is coming directly from `click`, so we'll either want to customize or wrap it.

File: src/globus_cli/commands/flows/run/update.py (hunk 1 above)
Comment (ada-globus) [reply src/globus_cli/commands/flows/run/update.py:21]: I don't have a very strong feeling, I just wanted to be sure I understood the intent. This is fine as-is, IMO.

**Pull request merged**